test-utils/assert/core/src/main/java/org/camunda/bpm/engine/test/assertions/bpmn/BpmnAwareTests.java

```diff
@@ -1013,7 +1039,29 @@ public static void complete(LockedExternalTask lockedExternalTask, Map<String, O
     if (lockedExternalTask == null || variables == null) {
       throw new IllegalArgumentException(format("Illegal call of completeExternalTask(lockedExternalTask = '%s', variables = '%s') - both must not be null!", lockedExternalTask, variables));
     }
-    externalTaskService().complete(lockedExternalTask.getId(), lockedExternalTask.getWorkerId(), variables);
+    complete(lockedExternalTask, variables, Collections.EMPTY_MAP);
+  }
+
+  /**
+   * Helper method to easily complete a locked external task.
+   *
+   * @param lockedExternalTask
+   *          an external task that was locked using the
+   *          {@link #fetchAndLock(String, String, int) fetchAndLock} method
+   * @param variables
+   *          Process variables to be passed to the process instance when
+   *          completing the task. For setting those variables, you can use
+   *          withVariables(String key, Object value, ...)
+   * @param localVariables
+   *          Local process variables to be passed to the process instance when
+   *          completing the task. For setting those variables, you can use
+   *          withVariables(String key, Object value, ...)
+   */
+  public static void complete(LockedExternalTask lockedExternalTask, Map<String, Object> variables, Map<String, Object> localVariables) {
+    if (lockedExternalTask == null || localVariables == null) {
+      throw new IllegalArgumentException(format("Illegal call of completeExternalTask(lockedExternalTask = '%s', localVariables = '%s') - both must not be null!", lockedExternalTask, localVariables));
+    }
```

Comment:
```suggestion
    if (lockedExternalTask == null || (variables == null && localVariables == null)) {
      throw new IllegalArgumentException(format("Illegal call of completeExternalTask(lockedExternalTask = '%s', variables = '%s', localVariables = '%s') - provide locked external task and either variables or local variables.", lockedExternalTask, variables, localVariables));
    }
```
Let's fail this explicitly when both `variables` and `localVariables` are `null`. Right now, when `variables` are provided, but `localVariables` are `null`, it would fail.